src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -1460,30 +1530,68 @@ void doDocument(DocumentManagement documentManagement,
             new Object[] { opentextDocId, fileDataSize });
         return;
     }
-    long versionNumber = version.getNumber();
-    String contextId = documentManagement.getVersionContentsContext(
-        documentNode.getID(), versionNumber);
-    ContentService contentService =
-        this.soapFactory.newContentService(documentManagement);
-    DataHandler dataHandler = contentService.downloadContent(contextId);
-
     String contentType = version.getMimeType();
     if (contentType != null) {
       response.setContentType(contentType);
     }
-    InputStream inputStream = dataHandler.getInputStream();
-    try {
-      if (inputStream != null) {
-        IOHelper.copyStream(inputStream, response.getOutputStream());
-      }
-    } finally {
-      if (inputStream != null) {
-        try {
-          inputStream.close();
-        } catch (IOException ioException) {
-          log.log(Level.FINE, "Error closing document stream", ioException);
+
+    switch (this.downloadMethod) {
+      case WEBSERVICES: {
+        log.log(Level.FINEST, "Downloading content from CWS");
+        long versionNumber = version.getNumber();
+        String contextId = documentManagement.getVersionContentsContext(
+            documentNode.getID(), versionNumber);
+        ContentService contentService =
+            this.soapFactory.newContentService(documentManagement);
+        DataHandler dataHandler = contentService.downloadContent(contextId);
+        try (InputStream inputStream = dataHandler.getInputStream()) {
+          IOHelper.copyStream(inputStream, response.getOutputStream());
```

Comment:
All the web services methods can return null (or at least aren't documented as never returning null) so I think I had just stuck with the pattern. The DataHandler doc makes it sound like it won't do that. I'll put the check back anyway.